core/src/retransmit_stage.rs

```diff
@@ -131,45 +132,29 @@ impl RetransmitStats {
     }
 }
 
-// Map of shred (slot, index, type) => list of hash values seen for that key.
-type ShredFilter = LruCache<ShredId, Vec<u64>>;
+struct ShredDeduper<const K: usize> {
+    deduper: Deduper<K, /*shred:*/ [u8]>,
```

Comment:
Won't these packets go through the `ShredFetchStage` deduper?

Reply:
The shreds recovered from erasure code are also piped through here, but they don't go through shred-fetch-stage.
So we will get a duplicate if a shred is recovered and then also arrives from turbine.